.github/workflows/build-test.yaml

```diff
@@ -62,12 +55,29 @@ jobs:
 
       - run: make deps test build/bin/kurl
 
+  test-shell:
```

Comment:
So, the test-shell must occurs after the the build-bin-kurl?
If so, should we not add here something like `needs: build-bin-kurl`?